packages/api/tests/lib/writeRecords/test-write-granules.js

```diff
@@ -513,18 +513,51 @@ test.serial('writeGranulesFromMessage() removes preexisting granule file from po
     knex,
     executionCumulusId,
     providerCumulusId,
-    granuleId,
   } = t.context;
 
+  // Set message status to 'completed' to allow file writes due to current file write constraints
+  cumulusMessage.meta.status = 'completed';
+
+  // Create granule in PG with multiple files. These records will exist in database
+  // during subsequent granule write from message
+  const files = [
+    fakeFileFactory({ size: 5 }),
+    fakeFileFactory({ size: 5 }),
+    fakeFileFactory({ size: 5 }),
+    fakeFileFactory({ size: 5 }),
+    fakeFileFactory({ size: 5 }),
+    fakeFileFactory({ size: 5 }),
+  ];
+  const existingGranule = fakeGranuleFactoryV2({
+    files: files,
+    granuleId: cryptoRandomString({ length: 10 }),
+    collectionId: constructCollectionId(t.context.collection.name, t.context.collection.version),
+  });
+  const existingPgGranule = await translateApiGranuleToPostgresGranule(existingGranule, knex);
+  const [existingPgGranuleRecordId] = await granulePgModel.create(knex, existingPgGranule, '*');
+
+  await Promise.all(files.map(async (file) => {
+    const pgFile = await translateApiFiletoPostgresFile(file);
+    pgFile.granule_cumulus_id = existingPgGranuleRecordId.cumulus_id;
+    return filePgModel.create(knex, pgFile);
+  }));
+  const existingPgFiles = await filePgModel.search(knex, {});
+
+  // Create the message granule and associated file in PG.
+  // The fakeFile created here is NOT in the message and will be deleted
+  // in writeGranulesFromMessage
   const pgGranule = await translateApiGranuleToPostgresGranule(granule, knex);
   const returnedGranule = await granulePgModel.create(knex, pgGranule, '*');
 
-  const fakeFile = await filePgModel.create(knex, {
+  const [fakeFile] = await filePgModel.create(knex, {
     granule_cumulus_id: returnedGranule[0].cumulus_id,
     bucket: 'fake_bucket',
     key: 'fake_key',
   }, '*');
 
+  // Ensure fakeFile was added to the files table
+  t.true(await filePgModel.exists(knex, { cumulus_id: fakeFile.cumulus_id }));
```

Comment:
Using `exists` here instead of the approach below:

```
t.deepEqual(updatedPgFiles.filter((file) => file.bucket === fakeFile.bucket), []);
```

It seemed easier and more explicit to do it this way.